### PR TITLE
chore: Intregrate runner packages.

### LINKED ISSUE #21689: Delete old runner code
## **Summary**
Once all the above is done and we are happy, we can remove a lot of old code:

* `packages/runner`
* `packages/runner-ct`
* Almost `packages/ui-components`, except the ones used in the Reporter
* probably `packages/runner-shared`, except a few files which can move to `packages/app`
* `LAUNCHPAD` environment variable - we will be permanently in unified mode now, so delete and associated conditionals with this
* anything that references `runner` and `runner-ct`

## **Acceptance Criteria**
* Should delete old code
* Should not break anything

┆Issue is synchronized with this [Jira Task](https://cypress-io.atlassian.net/browse/UNIFY-898) by [Unito](https://www.unito.io)
┆author: Lachlan Miller
┆epic: Chores
┆friendlyId: UNIFY-898
┆priority: Low
┆sprint: Backlog
┆taskType: Task

### PR DESCRIPTION
**Pull request by sainthkh**
- Related with #21689

### User facing changelog

N/A. It's an internal code cleanup.

### Additional details
- Why was this change necessary? => These are mostly-unused legacy code. 
- What is affected by this change? => N/A
- Any implementation details to explain? => I migrated `runner-shared` and `runner-ct` into `runner` and removed unused `scss` files and `react` components.

### Steps to test

`packages` folder structure has been changed. Run all tests to make sure that nothing has changed. 

### How has the user experience changed?

N/A. New contributors might not be confused with the legacy codes.

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/server/lib/controllers/runner.ts
Comment (lmiller1990) [inline packages/server/lib/controllers/runner.ts:95]: Nice, I wonder if we can just serve this from `runner-shared` instead? Ideally we should eventually eliminate both `runner` and `runner-ct` entirely.

In this case you'd probably need to change the webpack config to write the `dist` to `runner-shared`, or even move it there entirely. Thoughts?

File: .github/CODEOWNERS
Comment (lmiller1990) [inline .github/CODEOWNERS:45]: FYI some context - we ideally should delete ALL of runner/runner-shared, and bundle the entire app with the Vite stack.

The reason we are stuck like this is explained a little here: https://github.com/cypress-io/cypress/pull/20663#issuecomment-1073532399

Basically some of the code in driver and possibly reporter are not really ESM compatible - webpack does something different to Vite to resolve the modules, which is why it works in webpack, but we cannot import to Vite.

Not sure if this is something you are looking to investigate more @sainthkh but if you do please let me know, I'd like to follow along/help out - I'd really like to get away from this "inject webpack bundle" mess at some point, but it's quite hard. Likely making driver 100% ESM (including dependencies) would be a good first step.

**Review by lmiller1990 — COMMENTED** [review]
Left a few comments/questions, happy to approve once they've been replied too - nice job on this, great to see some of this getting cleaned up!

File: .gitignore
Comment (emilyrohrbough) [inline .gitignore:82]: delete. this is already included on line 40

**Review by emilyrohrbough — COMMENTED** [review]
We think alike! I love it. I started cleaning this up [last week](https://github.com/cypress-io/cypress/compare/clean-up-runner-pck) while waiting on test results because it was bugging me! I'll drop a few comments on things I noticed while cleaning this up.

File: packages/runner/package.json
Comment (emilyrohrbough) [inline packages/runner/package.json:24]: Need to add in the test script to run the mocha tests  that were in runner-shared

File: packages/server/lib/controllers/runner.ts
Comment (emilyrohrbough) [reply packages/server/lib/controllers/runner.ts:95]: I think we need to move towards removing the runner dirs all together. Ideally the app package would handle setting this up, even if both vite & webpack need to be used to bundle everything correctly. 🤷🏻‍♀️

File: packages/runner/package.json
Comment (emilyrohrbough) [inline packages/runner/package.json:43]: Can you update the runner's README as well?

It's doing a few more things than it currently lists with this merge:
- Contains `dom.js`, which uses proprietary webpack loaders and cannot easily be imported with Vite (dev server in `@packages/app`). Once `dom.js` is free of webpack-specific loader code, we should move it to `@packages/app`.
- Contains UI code for Cypress Studio, which was marked as experimental in Cypress 9.x and won't be part of Cypress 10.x initially. It will return at a later date. Until then, the code will be here. It's not currently used in the app.
- contains Legacy Cypress styles, most of these can likely be removed.

out of date:
- Some existing tests in `cypress/e2e` should be migrated to `@packages/app/cypress/e2e/runner`.

File: packages/runner-ct/webpack.config.ts
Comment (emilyrohrbough) [inline packages/runner-ct/webpack.config.ts:7]: We should add this to the webpack configuration in runner/webpack. I think it actually fixes #22404

File: packages/runner-ct/webpack.config.ts
Comment (emilyrohrbough) [inline packages/runner-ct/webpack.config.ts:75]: I want to call out this is the key / largest difference in webpack configuration difference between this and runner/webpack.config. I'll defer to the CT team if this is something that matters. if it does, I'd assume maybe here: https://github.com/cypress-io/cypress/pull/23028/files#diff-9af0f8f4f669446066e3dca23bb2645cb4f94dc9b11179a63f38cc657b36b30eR51

though assuming if tests are passing we are okay? cc// @lmiller1990 -- tagging you since you've already reviewed 😄

File: packages/server/lib/controllers/runner.ts
Comment (lmiller1990) [reply packages/server/lib/controllers/runner.ts:95]: Good point, there isn't really a compelling reason for `runner*` to exist, we may as well just move all the things to `app` (including webpack deps, config, etc). More packages just adds more misdirection.

File: .github/CODEOWNERS
Comment (sainthkh) [reply .github/CODEOWNERS:45]: It seems that we have 4 things to do to delete the `runner` pacakge.

- [ ] Migrate `dom`. -> I think `driver/src/dom` might be a good place or move it to `app` directly.
- [ ] Reimplement Studio -> #22870 
- [ ] ESM-ify `driver` -> I agree it's the biggest problem. 
- [ ] Let `reporter` compile its own styles. -> Move `runner/webpack.config.ts` to `reporter`.

File: packages/server/lib/controllers/runner.ts
Comment (sainthkh) [reply packages/server/lib/controllers/runner.ts:95]: When I read the current code, 

* `runner-ct` was dependent on `runner-shared` to use `dom` and `stduio`. 
* `runner` was dependent on `runner-ct` to use `unified-runner`.

So, the workflow became:

1. Migrate the `runner-shared` to `runner-ct`. (Remove unnecessary studio react files and some scss)
2. Migrate the `runner-ct` to `runner`. (Remove unnecessary scss files)

Now all the necessary codes of `runner-ct` and `runner-shared` are in `runner`.

File: .gitignore
Comment (sainthkh) [reply .gitignore:82]: Done.

File: packages/runner-ct/webpack.config.ts
Comment (lmiller1990) [reply packages/runner-ct/webpack.config.ts:75]: Yeah, I don't think we need this. We had some weird situation going on where we would conditionally load the react devtools, which we tried to embed in the runner-ct (which would be cool, if it worked as expected) but that experiment is long gone.

It sure would be nice to have the devtools embedded in the runner automatically... the main problem is the react devtools are designed to be used for 1 app and not torn down unless you refresh the page or via hot reload. When you run 100s of CT tests, we continually tear down/reset in AUT (and effectively the React app) every couple of seconds. Doing this leads to some situations where the devtools show stale components, or worse, keep a reference to components that no longer exist ... which can give us a memory leak.

File: packages/runner/package.json
Comment (sainthkh) [reply packages/runner/package.json:43]: Done. I changed the `UI code for Cypress Studio` with `Cypress Studio Recorder code`. Because I removed `react` UI code in this PR and `Studio Recorder` is currently injected into `app`.

File: .github/CODEOWNERS
Comment (lmiller1990) [reply .github/CODEOWNERS:45]: Dom -> are you going do to this here?
Studio -> I am looking into this now
ESMify driver -> agree, hard
Reporter styles -> do this separately I think?

**Review by lmiller1990 — APPROVED** [review]
Nice cleanup! Going to approve, but also happy to re-review if you move the DOM code to `app`.

Nice list of TODOs here: https://github.com/cypress-io/cypress/pull/23028#discussion_r936142417

File: .github/CODEOWNERS
Comment (sainthkh) [reply .github/CODEOWNERS:45]: I read the `dom` code and it seems that it's better to be a separate PR.

It currently renders highlight by

1. Creating the container node with jQuery.
2. Rendering the Highlight DOM with React.

I can simply migrate `dom` to `app` and use `unified-runner`. But I don't think it's not our goal. It's just hiding and postponing. 

To get migrating `dom` done, I need to do these things:

- [ ] Remove JQuery from `dom` and `dimensions`.
- [ ] Move the current JQuery version of `dimensions` to `driver/src/dom` because it's used in `driver/src/dom/blackout`.
- [ ] Migrate `Highlight` from `react` to `vue` with our `Tooltip` component. 
- [ ] Migrate `dom` functions to `app/runner/aut-iframe`. Leave `studio`-related functions.
- [ ] Clean up unified-runner `dom` and `CypressJQuery`. They're only used inside `aut-iframe`. I guess they can be removed safely from `unified-runner`.

I think it's a big change and outside the scope of this PR. 

----

As for reporter styles, it's safe to do it after ESM-ifying `driver`. We're trying to remove `webpack`. I don't think we don't need to create one more `webpack.config.ts` now.

File: packages/runner/package.json
Comment (sainthkh) [reply packages/runner/package.json:24]: Currently, `runner-shared` unit tests are ignored. 

https://github.com/cypress-io/cypress/blob/f69afb5aa49996cace863f8a3ae142698deff982/package.json#L53

And it fails in the current `develop` branch.

And the only the file that exists is `studio-recorder.spec.js`. It's there for reference purpose and it'll be removed soon with #22870.

----

I checked the code base and removed `network`, `socket`, and `rewriter`.

File: packages/runner-ct/webpack.config.ts
Comment (sainthkh) [reply packages/runner-ct/webpack.config.ts:7]: I fixed this and learned that favicon shows when the URL port is `4455`, but it fails when `5555`.

File: .github/CODEOWNERS
Comment (sainthkh) [reply .github/CODEOWNERS:45]: I opened the new PR #23187 to migrate `dom.js`. It's currently empty because I'm now reading and planning.

**Comment by lmiller1990** [comment]
Are you still working on this @sainthkh or are we good to re-review and merge if green?

**Comment by sainthkh** [comment]
@lmiller1990  I think it's done. I'll take care of the `dom` issue at #23187.

**Review by emilyrohrbough — APPROVED** [review]

**Comment by lmiller1990** [comment]
run-reporter seems broken. Last blocker @sainthkh :thinking:

**Comment by sainthkh** [comment]
@lmiller1990  It's happening because there are 4 test files inside `reporter`(`commands/command.cy.tsx`, `hooks/hooks.cy.tsx`, `lib/tag.cy.tsx`, `test/test.cy.tsx`), but there are 7 machines to run them. 3 of them don't have work to do. I opened an issue about this misleading message a few months ago at #22328.

I intentionally didn't fix this problem at #22326 because I thought the problem is more at the ci setting side [(comment)](https://github.com/cypress-io/cypress/pull/22326#issuecomment-1157295392).

There are a few solutions for this problem:

1. Change the CI setting from 7 to 2 or 3. -> I think it's better because we need less power and it's unlikely that we might need more than 2 or 3 machines for the tests.
2. Return nothing when there is no test file. -> We need to fix the ci script. Maybe it's a different issue and open a new PR.

**Comment by lmiller1990** [comment]
Let's just take it down to `parallelism: 2` for now. For such a small number of specs, we probably don't need it at all...

**Pull request merged**
